website/docs/crud.md

```diff
@@ -0,0 +1,294 @@
+---
+id: crud
+title: CRUD Handler
+sidebar_label: CRUD Handler
+---
+
+## CRUD
+In `Core-Lib`, we have classes that can handle CRUD  ( Create, Read, Update, Delete ) for us.
+They are as follows
+
+
+
+### CRUD
+This class is used to initialize `CRUD` with database objects, rule validators and database handlers. It is the Base
+class for other classes responsible for CRUD
+```python
+class CRUD(ABC):
+
+    def __init__(self, db_entity, db: SqlAlchemyDataHandlerRegistry, rule_validator: RuleValidator):
+        self._db_entity = db_entity
+        self._db = db
+        self._rule_validator = rule_validator
+```
+`db_entity` is the database class or object to be passed for initialization.
+
+`db` is the `SqlAlchemyDataHandlerRegistry` instance used to connect to the database.
+
+`rule_validator` rules must be created with `RuleValidator` and passed.
+
+#### Functions provided by `CRUD`
+
+- `get()` is overridden by other CRUD classes for implementing the function as per their needs.
+
+
+- `delete()` is overridden by other CRUD classes for implementing the function as per their needs.
+
+
+- `create()` is used to add a new entry to the database, this function takes a `dict` with the data to be added to the database.
+```python
+def create(self, data: dict):
+```
+`data` is type `dict`, key-values pair where key is the column name and value is the entry to be added to the column.
+
+
+- `update()` is used to update a column in database, This function takes a `dict` that contains the data to be updated 
+as well as the `id` of the column that needs to be updated.
+```python
+def update(self, id: int, data: dict):
+```
+`id` is type `int`, takes the id of the column to be updated.
+
+`data` is type `dict`, key-values pair where key is the column name and value is the entry to be updated.
+
+### CRUDDataAccess
+This class is used to access data via `CRUD` and we have to extend this class to use `CRUD` in your application. This

```

Comment:
simply say


extends the `CRUD` and implementing the `get` and `delete` methods